src/history/hs_rec.c

```diff
@@ -855,6 +835,17 @@ __hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, ui
         return (0);
     WT_ERR(ret);
 
+    /*
+     * Fail the eviction if we detect out of order timestamp when checkpoint is running. We cannot
+     * modify the history store to fix the out of order timestamp updates as it may make the history
+     * store checkpoint inconsistent.
+     */
+    if (checkpoint_running) {
```

Comment:
I am curious if there is a better way of detecting a running checkpoint. What are the implications if we use `WT_TXN_GLOBAL::checkpoint_running` flag rather than passing around the flag?

Reply:
Or we want to check if the checkpoint running when starting evicting the page? In that case we have to keep this flag I believe.